src/tools/westcos-tool.c

```diff
@@ -678,6 +678,10 @@ int main(int argc, char *argv[])
 		}
```

Comment:
`dst->modulus` already leaks if the condition on line 675 fails. It would be worth adding the free there too.